### ISSUE TITLE
Fix batch job death with htex

### ISSUE THREAD

**Issue by annawoodard** [Parsl/parsl#633]
The following test script gets to ~4k tasks completed, then the slurm job dies. This probably requires a fix for #631 to properly debug. Note that it seems to work after changing `cores_per_worker=2`.

```
from requests import get
import parsl
from parsl.app.app import python_app

from parsl.executors import HighThroughputExecutor
from parsl.config import Config
from parsl.app.app import python_app
from parsl.providers.slurm.slurm import SlurmProvider
from parsl.launchers import SimpleLauncher


config = Config(
    executors=[
        HighThroughputExecutor(
            label="midway_htex",
            worker_debug=True,
            cores_per_worker=1,
            public_ip=get('https://api.ipify.org').text,
            provider=SlurmProvider(
                'broadwl',
                launcher=SimpleLauncher(),
                overrides='source ~/.bashrc; source activate parsl',
            ),
        )
    ],
    strategy=None,
)

parsl.set_stream_logger()
parsl.load(config)

@python_app
def noop():
    pass

for trial in range(10):
    tasks = [noop() for _ in range(0, 1000)]
    [t.result() for t in tasks]
```

**Comment by yadudoc** [comment]
@annawoodard Can you try this again with branch: version_mismatch_#244

**Comment by yadudoc** [comment]
To better understand this I ran a barebones ZMQ test on midway: I set up a ZMQ server with a ROUTER socket on swift.rcc.uchicago.edu. I got an interactive node and ran a simple ZMQ client with a DEALER socket to connect and send a numbered message every second. The connection gets dropped consistently after 19s which is the behavior I see with the worker connections with the interchange. The login node has a two internal networks and when using those interfaces this issue vanishes.

I have not been able to fully diagnose the root cause of the behavior I'm seeing on midway, however now that we have a reasonable alternative (use internal network addresses) I do not think this should be blocking 0.7.0.
